controllers/serviceaccount_controller_unit_test.go

```diff
@@ -54,9 +54,6 @@ func unitTestsReconcileNormal() {
 		_, err := reconciler.ReconcileNormal(ctx, controllerCtx.GuestClusterContext)
 		Expect(err).NotTo(HaveOccurred())
 
-		// Update the VSphereCluster and its status in the fake client.
-		Expect(controllerCtx.ControllerContext.Client.Update(ctx, controllerCtx.VSphereCluster)).To(Succeed())
-		Expect(controllerCtx.ControllerContext.Client.Status().Update(ctx, controllerCtx.VSphereCluster)).To(Succeed())
```

Comment:
These functions weren't being used in the underlying test as the object was being checked in place. Some change in the fakeclient seems to have changed how `Update` works so that e.g. `Status` update also mutates the object in place making this function not work. Fine to get rid of it here, and there weren't that many changes needed for this bump.

I do wonder with how often `Update` functions are used across the code what kind of bugs are being missed though.

Reply:
Q: "being checked in place" this means our reconciler code is just reading it from our magic ctx and not via the client. So there is no need for the VSphereCluster object to exist in the fakeclient, right?

Reply:
Exactly